### DISCUSSION TITLE
web analytics / tag manager

### DISCUSSION THREAD

**Discussion by marco-santulli** [Budibase/budibase#3869] [Ideas]
Hi,

Is there an easy way to add a tag manager or other analytics to the budibase pages? wouldn't it be useful?

thanks

Marco

**Comment by marco-santulli** [comment]
bump

**Reply by mjashanks** [reply]
oh, hey Marco. Missed this. AFAIK there is no easy way to do this, and we don't have a place on the roadmap for the near future (tho I agree it would be useful).

Have you tried using an `Embed` component, with a `<script>` inside?

**Reply by jimbobcraig** [reply]
Taking this approach didn't work for me. I'm trying to use a single-line [Umami](https://umami.is/) script for tracking visitor numbers to a public form. I'd be interested to hear if there are any other ideas for a solution.

**Comment by marco-santulli** [comment]
Hi Mike,

no worries, that sounds like a possibility, I will give it a shot.  Maybe it could be easy to do if you ad a checkbox or something on the "embed" component telling budi to put it at the end of the header?

M

**Reply by hafbau** [reply]
did this work?

**Comment by hreinberger** [comment]
So I got around to adding an Analytics solution to my Budibase Instance. For future reference here's how I have done it.

1. I run Cloudflare as DNS and Cache for the site. They have just announced a new beta tool called "[Zaraz](https://developers.cloudflare.com/zaraz/)" that injects arbitrary code into the site on runtime from the edge. Scary, but useful.
2. I added a new "custom html" tool in there that loads my [Plausible](https://plausible.io/) tracking JS on every page load.
3. Then I edited the [nginx.conf](https://github.com/Budibase/budibase/blob/develop/hosting/nginx.prod.conf.hbs) to allow loading and connecting to the script from the [CSP](https://github.com/Budibase/budibase/blob/596f27740d4e8caa6a2945d7cd4d89be4a0a9394/hosting/nginx.prod.conf.hbs#L47)
4. I then mounted the modified conf into the production container just like [here](https://github.com/Budibase/budibase/blob/596f27740d4e8caa6a2945d7cd4d89be4a0a9394/hosting/docker-compose.dev.yaml#L29).

Minor inconvenience: Since the tracking JS is injected at every page load, it also tracks actions in the backend. You can create "page rules" in cloudflare to disable Zaraz on certain pages though. I just added a rule that disables Zaraz on `mysite.com/builder/*`

Yes, you'll have to jump through some hoops and this may or may not apply to your Tracking app and infrastructure, but I wanted to document this _somewhere_.

**Comment by joebudi** [comment]
Budibase Enterprise now supports injecting custom scripts directly into your applications. This feature enables advanced customization and integration with third-party libraries and services, including analytics.

[Budibase now all](https://docs.budibase.com/changelog/app-scripts)
